cli/internal/runsummary/run_summary.go

```diff
@@ -124,10 +116,16 @@ func NewRunSummary(
 		repoRoot:           repoRoot,
 		singlePackage:      singlePackage,
 		shouldSave:         shouldSave,
-		apiClient:          apiClient,
-		spaceID:            spaceID,
 		synthesizedCommand: synthesizedCommand,
 	}
+
+	// Note: this sets up a bidirectional relationship
+	// rsm has a reference to the spacesClient, and spacesClient has a reference to rsm
+	rsm.spacesClient = newSpacesClient(spaceID, apiClient, ui, &rsm)
```

Comment:
FWIW it looks like you don't need to create the circular relationship. You can pass `rsm` in in the place you're using it.